src/goto-checker/report_util.cpp

```diff
@@ -198,7 +199,11 @@ get_sorted_properties(const propertiest &properties)
       const auto &p2 = pit2->second.pc->source_location;
       if(p1.get_file() != p2.get_file())
         return id2string(p1.get_file()) < id2string(p2.get_file());
-      else if(!p1.get_line().empty() && !p2.get_line().empty())
+      if(p1.get_function() != p2.get_function())
+        return id2string(p1.get_function()) < id2string(p2.get_function());
+      else if(
+        !p1.get_line().empty() && !p2.get_line().empty() &&
+        p1.get_line() != p2.get_line())
         return std::stoul(id2string(p1.get_line())) <
                std::stoul(id2string(p2.get_line()));
       else
```

Comment:
I think it’d probably be better to sort ids numerically (by the number at the end) rather than lexicographically here

Reply:
Done.